metadata-ingestion/src/datahub/ingestion/source/sql/sql_utils.py

```diff
@@ -235,3 +236,24 @@ def schema_requires_v2(canonical_schema: List[SchemaField]) -> bool:
         if ARRAY_TOKEN in field_name or UNION_TOKEN in field_name:
             return True
     return False
+
+
+def check_table_with_profile_pattern(
+    profile_pattern: AllowDenyPattern, table_name: str
+) -> bool:
+    parts = len(table_name.split("."))
+    allow_list: List[str] = []
+
+    for pattern in profile_pattern.allow:
+        replaced_pattern = pattern.replace(".*", "").replace(".+", "")
+        splits = re.split(r"\\?\.", replaced_pattern)
```

Comment:
we should cache the conversion of profile_pattern -> table_allow_deny_pattern, since the regex stuff might be expensive